reconcile/utils/saasherder.py

```diff
@@ -206,14 +218,77 @@ def _validate_saas_files(self):
             for saas_file_name, saas_file_paths in duplicates.items():
                 logging.error(msg.format(saas_file_name, saas_file_paths))
 
-        # promotion publish channel duplicates
-        duplicates = [p for p in saas_file_promotion_publish_channels
-                      if saas_file_promotion_publish_channels.count(p) > 1]
-        if duplicates:
-            self.valid = False
-            msg = 'saas file promotion publish channel is not unique: {}'
-            for duplicate in duplicates:
-                logging.error(msg.format(duplicate))
+        self._check_promotions_have_same_source(subscriptions, publications)
+
+    def _get_promotion_pubs_and_subs(
+            self,
+            rt_ref: Tuple,
+            promotion: dict[str, Any],
+            publications: MutableMapping[str, Tuple],
+            subscriptions: MutableMapping[str, list[Tuple]]):
+        """
+        Function to gather promotion publish and subcribe configurations
+        It validates a publish channel is unique across all publis targets.
+        """
+        publish = promotion.get('publish')
+        if publish:
```

Comment:
I think we can safe one indention here. A neat trick I recently learned from Maor :)

```
publish = promotion.get('publish') or []
for channel in publish:
```

Reply:
changed. thanks